examples/bite_sized/dynamic_output.py

```diff
@@ -0,0 +1,72 @@
+import csv
+import re
+import os
+import xml.etree.ElementTree as ET
+
+import requests
+import pandas as pd
+
+from dagster import (
+    Out,
+    Output,
+    job,
+    op,
+    resource,
+    DynamicOut,
+    DynamicOutput,
+    io_manager,
+    OutputContext,
+)
+
+from bite_sized.conditional_branching import DataframeToCSVIOManager
+
+ARTICLES_LINK = "https://rss.nytimes.com/services/xml/rss/nyt/HomePage.xml"
+
+
+class DynamicOutIOManager(DataframeToCSVIOManager):
+    def _get_path(self, output_context: OutputContext):
+        path = f"{output_context.step_key}{f'_{output_context.mapping_key}' if output_context.mapping_key else ''}.csv"
+        return os.path.join(
+            self.base_dir,
+            path,
+        )
+
+
+@io_manager(config_schema={"base_dir": str})
+def dynamic_df_to_csv_io_manager(init_context):
+    return DynamicOutIOManager(base_dir=init_context.resource_config.get("base_dir"))
+
+
+@op(out=DynamicOut())
+def fetch_articles_df():
+    tree = ET.fromstring(requests.get(ARTICLES_LINK).text)
+
+    articles_by_category = {}
+
+    for article in tree[0].findall("item"):
+        for category in article.findall("category"):
+            category_name = category.text
+            if category_name not in articles_by_category:
+                articles_by_category[category_name] = []
+            articles_by_category[category_name].append(
+                {
+                    "Title": article.find("title").text,
+                    "Link": article.find("link").text,
+                    "Category": category_name,
+                    "Description": article.find("description").text,
+                }
+            )
+
+    for category_name, articles in articles_by_category.items():
+        yield DynamicOutput(pd.DataFrame(articles), mapping_key=re.sub(r"\W+", "", category_name))
+
+
+@op
+def combine_articles_df(articles):
+    return pd.concat(articles)
```

Comment:
would be great to have some more parsing logics here to make "why" to use the dynamic out stronger.

the current job shape imo could be simplified to a single op that fetch the df and store per-category in csv's a for loop, i.e. it's a bit less convincing why i need to have two ops and dynamic out at the first place.